jwk/helper.go

```diff
@@ -120,6 +127,25 @@ func FindKeyByPrefix(set *jose.JSONWebKeySet, prefix string) (key *jose.JSONWebK
 	return First(keys.Keys), nil
 }
 
+func FindPublicKey(set *jose.JSONWebKeySet) (key *jose.JSONWebKey, err error) {
+	keys := ExcludePrivateKeys(set)
+	if len(keys.Keys) == 0 {
+		return nil, errors.New("key not found")
+	}
+
+	return First(keys.Keys), nil
+}
+
+func ExcludePrivateKeys(set *jose.JSONWebKeySet) *jose.JSONWebKeySet {
+	keys := new(jose.JSONWebKeySet)
+	for _, k := range set.Keys {
+		if _, ok := k.Key.(*rsa.PrivateKey); !ok {
+			keys.Keys = append(keys.Keys, k)
+		}
+	}
+	return keys
+}
+
```

Comment:
Could you please add a test for these two functions which:

1. Generate an RSA private/public keypair
2. Generate an ECDSA private/public keypair
3. Run `FindPublicKey` and `ExcludePrivateKeys`
4. Expect no private keys to be exposed, but public keys to be exposed